template/addons/next-auth/protected-page.tsx

```diff
@@ -0,0 +1,26 @@
+import type { NextPage, GetServerSidePropsContext } from "next";
+import { signOut } from "next-auth/react";
+import { getServerAuthSession } from "../server/common/get-server-auth-session";
+
+const ProtectedPage: NextPage = () => {
```

Comment:
Definitely, gonna add some comments sugesting an usual flow on how to get the session data and use it on the page

Reply:
Also will try to use the it in the places will suggest it. One little problem i came up with was with types, because the `getServerAuthSession` expects an object with `req` and `res` that are `GetServerSidePropsContext["req"]` and ``GetServerSidePropsContext["res"]``, but in some places (like the restricted endpoint withou tRPC) uses `req` and `res` as `NextApiRequest` and `NextApiResponse`.

I've just ignored it for the restricted enpoint because i think the `getServerAuthSession` was mainly proposed to be used in gSSP, but i will try in this files you pointed

Reply:
Oh wow are Next using different types for gSSP and API handlers...

Does it work for both? Like can we do something like
```ts
export const getServerAuthSession = async (
  ctx:
    | {
        req: GetServerSidePropsContext["req"];
        res: GetServerSidePropsContext["res"];
      }
    | { 
        req: NextApiRequest; 
        res: NextApiResponse;
      }
) => {
  return await unstable_getServerSession(ctx.req, ctx.res, nextAuthOptions);
};
```

Reply:
Actually i've tried here and worked hahah my bad

Already commited!